### DISCUSSION TITLE
Why can't I solve this simple convex integer program?

### DISCUSSION THREAD

**Discussion by erelsgl** [cvxpy/cvxpy#2429] [Q&A]
When I run this code:
```
x = cvxpy.Variable(integer=True)
y = cvxpy.Variable(integer=True)
objective = cvxpy.Minimize((x - y)**2)
constraints = [x + y == 5,  1 <= x, x <= 3,    y >= 1, 3 >= y]  
prob = cvxpy.Problem(objective, constraints)
prob.solve()
```

I get the following error:
```
SolverError: Either candidate conic solvers (['GLPK_MI', 'CBC', 'SCIPY']) do not support the cones output by the problem (SOC, NonNeg, Zero), or there are not enough constraints in the problem.
```

What is the problem with my problem? Why can't it be solved?

**Comment by phschiele** [comment]
Please refer to the solvers to see which cones they can handle:
https://www.cvxpy.org/tutorial/advanced/index.html#choosing-a-solver

You seem to have the MIP capable solvers `['GLPK_MI', 'CBC', 'SCIPY']`, but none of them support the quadratic / SOC objective. The problem will be solvable as specified with Gurobi, Mosek, SCIP, XPRESS, and CPLEX.
